daemon/cmd/kube_proxy_replacement.go

```diff
@@ -238,6 +238,27 @@ func initKubeProxyReplacementOptions() (strict bool) {
 		}
 	}
 
+	if !option.Config.EnableHostLegacyRouting {
+		switch {
+		case !option.Config.EnableNodePort:
+			fallthrough
+		case option.Config.Tunnel != option.TunnelDisabled:
+			fallthrough
+		// Non-BPF masquerade requires netfilter and hence CT.
+		case option.Config.Masquerade && !option.Config.EnableBPFMasquerade:
+			option.Config.EnableHostLegacyRouting = true
```

Comment:
Considering the lacking multi-dev support, shall we currently switch back to the legacy routing when more than one NodePort device is detected/set?

Reply:
I think in the mult-dev support case checking the ct.state->ifindex should be enough. If it exists we should know where to send the packet if it doesn't exist we can simply wait for it to be populated by passing packet to stack and using the "slowpatch" linux case. It should just be the first syn packet. So still good for better established flows, I guess its not ideal for short single packet send/recv flows. But, might be easy to implement compared with doing the lookup.

Reply:
(as discussed via sig-dp, ct.state->ifindex can be used for caching)

Reply:
How do we know in this path that `bpf_redirect_neigh` is supported?